nixos/modules/programs/environment.nix

```diff
@@ -25,7 +25,8 @@ in
 
     environment.sessionVariables =
       { NIX_PATH =
-          [ "/nix/var/nix/profiles/per-user/root/channels/nixos"
+          [ "/nix/var/nix/profiles/per-user/root/channels"
+            "/nix/var/nix/profiles/per-user/root/channels/nixos"
```

Comment:
A slight problem here is that you end up with an ambiguous/unintended situation if you add both the Nixpkgs and NixOS channel, because then NixOS will be updated from the Nixpkgs rather than NixOS channel.